Vostok.ClusterClient.Core/Modules/IRequestContext.cs

```diff
@@ -39,10 +39,16 @@ public interface IRequestContext
         ILog Log { get; }
 
         /// <summary>
-        /// Gets or sets ClusterProvider instance used to send request.
+        /// Gets or sets <see cref="ClusterProvider"/> instance used to send request.
         /// </summary>
         [NotNull]
         IClusterProvider ClusterProvider { get; set; }
+        
+        /// <summary>
+        /// Gets or sets <see cref="AsyncClusterProvider"/> instance used to send request.
+        /// </summary>
+        [CanBeNull]
+        IAsyncClusterProvider AsyncClusterProvider { get; set; }
```

Comment:
I hope no one implementing their own `IRequestContext` and this extensions wouldn't broke anything.